src/crl/interactivesessions/shells/bashshell.py

```diff
@@ -77,13 +77,13 @@ def _set_bash_environment(self):
         logger.log(_LOGLEVEL, "Preparing bash session")
         output = self._read_until_end()
 
-        self._send_input_line('unset HISTFILE')
+        self._send_input_line('unset HISTFILE', remove_from_buffer=False)
         stty_cmd = "stty cols 400 rows 400 {0}".format(
             'echo' if self.tty_echo else '-echo')
-        self._send_input_line(stty_cmd)
-        self.set_prompt()
+        self._send_input_line(stty_cmd, remove_from_buffer=False)
```

Comment:
This really does not make sense at all. We have already set via `stty` command the `termios` echo to be consistent with the `self.tty_echo`. Now, we override this here by forcing essentially `self.tty_echo=False`.